shaders/program/shadowcomp1.glsl

```diff
@@ -422,30 +438,37 @@ void main() {
                         weight += otherWeight;
                     }
                     GILight /= weight;
-                    for (int k = 0; k < 3; k++) {
-                        normal[k] = getDistanceField(vxPos + mat3(0.5)[k]) - getDistanceField(vxPos - mat3(0.5)[k]);
-                    }
-                    normal = normalize(normal);
                     vxPos -= min(0.3, thisDFval - 0.1) * normal;
                     for (int sampleNum = 0; sampleNum < GI_SAMPLE_COUNT; sampleNum++) {
                         vec3 dir = randomSphereSample();
                         if (dot(dir, normal) < 0.0) dir = -dir;
                         float ndotl = dot(dir, normal);
                         vec3 hitPos = rayTrace(vxPos, LIGHT_TRACE_LENGTH * dir, dither);
-                        vec3 hitCol = vec3(0);
+                        #ifdef GL_CAVE_FACTOR
+                            vec3 hitCol = ambientColor * clamp(dir.y + 1.6, 0.6, 1) * (1-GetCaveFactor(cameraPosition.y + vxPos.y)) / GI_STRENGTH;
```

Comment:
I think I store skylight somewhere in the voxel data, could probably use that. will check where it is and try if it looks alright

Reply:
Never mind that, it would look bad

Reply:
IMO Caves look decent rn, but xlarge builds still have light coming from nowhere. Maybe could be improved with skylight, but dont want to affect other situations too much